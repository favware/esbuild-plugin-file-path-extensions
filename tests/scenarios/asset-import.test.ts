import esbuild, { type BuildOptions } from 'esbuild';
import { assertFileContent, buildAbsolutePath, createEsbuildConfig } from '../common/util';

function createConfig(): BuildOptions {
  return createEsbuildConfig({
    platform: 'node',
    format: 'esm',
    outExtension: {
      '.js': '.mjs'
    },
    entryPoints: [buildAbsolutePath('../fixtures/build-in/asset.ts')]
  });
}

describe('Importing assets', () => {
  test('GIVEN an asset import THEN leaves path alone', async () => {
    const config = createConfig();

    await esbuild.build(config);

    await assertFileContent('../fixtures/build-out/asset.mjs');
  });
});
