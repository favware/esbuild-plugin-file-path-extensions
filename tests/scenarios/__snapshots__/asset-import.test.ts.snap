// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Importing assets > GIVEN an asset import THEN leaves path alone 1`] = `
"// tests/fixtures/build-in/asset.ts
import "./asset.css";
import styles from "./asset.module.css";
import json from "./asset.json";
function test() {
  console.log("styles", styles);
  console.log("json", json);
}
export {
  test
};
"
`;
